src/TodoContext/TodosContext.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useMemo, useState } from 'react';
+import { Todo } from '../Interface/Todo';
+import { Status } from '../Enum/Status';
+import { useLocalStorage } from '../hooks/UseLocalStorage';
+
+type TodoContext = {
+  todos: Todo[],
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>
+  toggleAllStatus: boolean,
+  setToggleAllStatus: (status: boolean) => void,
+  setFilterBy: (value: Status) => void,
+  filterBy: Status,
+  todosFilter: () => Todo[],
+};
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const TodosContext = React.createContext<TodoContext>({
+  todos: [],
+  setTodos: () => {},
+  toggleAllStatus: false,
+  setToggleAllStatus: () => {},
+  setFilterBy: () => {},
+  filterBy: Status.ALL,
+  todosFilter: () => [],
+});
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useLocalStorage('todos', []);
+  const [toggleAllStatus, setToggleAllStatus] = useState<boolean>(false);
```

Comment:
Fix all boolean variable naming using [tips](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)
```suggestion
  const [isToggleAllStatus, setIsToggleAllStatus] = useState<boolean>(false);
```